game-core/src/main/java/games/strategy/triplea/ui/menubar/LobbyMenu.java

```diff
@@ -201,23 +200,24 @@ private static INode newDummyNode(final String name) {
   private void addBanMacAddressMenuItem(final JMenu parentMenu) {
     final JMenuItem menuItem = new JMenuItem("Ban Hashed Mac Address");
     menuItem.addActionListener(e -> {
-      final @Nullable String hashedMacAddress = showInputDialog(
-          "Enter the hashed Mac Address that you want to ban from the lobby.\n\n"
-              + "Hashed Mac Addresses should be entered in this format: $1$MH$345ntXD4G3AKpAeHZdaGe3");
-      if (validateHashedMacAddress(hashedMacAddress)) {
-        showTimespanDialog(
-            "Please consult other admins before banning longer than 1 day.",
-            date -> getModeratorController().banMac(newDummyNode("__unknown__"), hashedMacAddress, date));
-      }
+      showHashedMacAddressInputDialog("Enter the hashed Mac address that you want to ban from the lobby.")
+          .filter(this::validateHashedMacAddress)
+          .ifPresent(hashedMacAddress -> showTimespanDialog(
+              "Please consult other admins before banning longer than 1 day.",
+              date -> getModeratorController().banMac(newDummyNode("__unknown__"), hashedMacAddress, date)));
     });
     parentMenu.add(menuItem);
   }
 
+  private Optional<String> showHashedMacAddressInputDialog(final String message) {
+    final String completeMessage = message + "\n"
+        + "\n"
+        + "Hashed Mac addresses should be entered in this format: $1$MH$345ntXD4G3AKpAeHZdaGe3";
+    return showInputDialog(completeMessage).map(MacFinder::withPrefix);
```

Comment:
This is the functional change that automatically adds the hashed MAC address prefix, if needed.